File: java/api/src/main/java/net/lab1318/costume/api/models/institution/InstitutionId.java
package net.lab1318.costume.api.models.institution;

import org.notaweb.api.models.StringModelId;

public final class InstitutionId extends StringModelId {
    public static InstitutionId parse(final String value) throws InvalidInstitutionIdException {
        if (value.isEmpty()) {
            throw new InvalidInstitutionIdException("empty string");
        }
        return new InstitutionId(value);
    }

    private InstitutionId(final String value) {
        super(value);
    }
}

File: thrift/src/costume/api/models/institution/institution_id.py
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from _id import _Id


class InstitutionId(_Id):
    def java_read_protocol_throws_checked(self):
        return ['net.lab1318.costume.api.models.institution.InvalidInstitutionIdException']

    def java_qname(self, boxed=False):
        return 'net.lab1318.costume.api.models.institution.InstitutionId'
